include/aikido/planner/ompl/OMPLConfigurationToConfigurationPlanner.hpp

```diff
@@ -0,0 +1,110 @@
+template <class PlannerType>
+class OMPLConfigurationToConfigurationPlanner
+    : public aikido::planner::ConfigurationToConfigurationPlanner
+{
+public:
+  /// Constructor.
+  ///
+  /// \param[in] planner The OMPL planner to use.
+  /// \param[in] stateSpace State space that this planner associated with.
+  /// \param[in] rng Random number generator to create the state sampler.
+  /// \param[in] interpolator Interpolator used to interpolate between two
+  /// states. GeodesicInterpolator is used by default.
+  /// \param[in] dmetric A valid distance metric defined on the StateSpace.
+  /// Distance metric relevant to the statespace is used by default.
+  /// \param[in] sampler A Sampleable that can sample states from the
+  /// StateSpace. Warning: Many OMPL planners internally assume this sampler
+  /// samples uniformly. Care should be taken when using a non-uniform sampler.
+  /// \param[in] boundsConstraint A constraint used to determine whether states
+  /// encountered during planning fall within any bounds specified on the
+  /// StateSpace. In addition to the validityConstraint, this must also be
+  /// satisfied for a state to be considered valid.
+  /// \param[in] boundsProjector A Projectable that projects a state back within
+  /// valid bounds defined on the StateSpace.
+  /// \param[in] maxDistanceBtwValidityChecks The maximum distance (under
+  /// dmetric) between validity checking two successive points on a tree
+  /// extension or an edge in a graph.
+  OMPLConfigurationToConfigurationPlanner(
+      statespace::ConstStateSpacePtr stateSpace,
+      common::RNG* rng = nullptr,
+      statespace::ConstInterpolatorPtr interpolator = nullptr,
+      distance::DistanceMetricPtr dmetric = nullptr,
+      constraint::SampleablePtr sampler = nullptr,
+      constraint::TestablePtr boundsConstraint = nullptr,
+      constraint::ProjectablePtr boundsProjector = nullptr,
+      double maxDistanceBtwValidityChecks = 0.1);
+
+  /// Plans a trajectory from start state to goal state by using an interpolator
+  /// to interpolate between them.
+  ///
+  /// The planner returns success if the resulting trajectory satisfies
+  /// constraint at some resolution and failure (returning \c nullptr)
+  /// otherwise. The reason for the failure is stored in the \c result output
+  /// parameter.
+  ///
+  /// \param[in] problem Planning problem.
+  /// \param[out] result Information about success or failure.
+  /// \return Trajectory or \c nullptr if planning failed.
+  /// \throw If \c problem is not ConfigurationToConfiguration.
+  /// \throw If \c result is not ConfigurationToConfiguration::Result.
+  trajectory::TrajectoryPtr plan(
+      const SolvableProblem& problem, Result* result = nullptr) override;
+
+  /// Returns the underlying OMPL planner used.
+  ::ompl::base::PlannerPtr getOMPLPlanner();
+
+  /// Sets interpolator used to produce the output trajectory.
+  void setInterpolator(statespace::ConstInterpolatorPtr interpolator);
```

Comment:
Nit: missing documentation for `interpolator`.

Reply:
Is this function being used? What's the use case for wanting to change the interpolator after construction, without changing the other fields?